Add universal Link content type (#6134)
sulu/sulu: src/Sulu/Bundle/AdminBundle/Resources/js/containers/Link/README.md

> The `Link` component is a universal link selection and can additionally also be used for external links.

Reviewer: ```suggestion
The `Link` component implements a universal link selection that allows to select internal and external links.
It can be extended with additional custom link types via the `linkTypeRegistry`.

```

The component displays a `Select` together with an input field. The `select` component is used to choose the type

> (page, media, external, article). The input field renders the currently selected entity or the external url.

Reviewer: ```suggestion
(eg. `page`, `media` or `external`). The input field renders the currently selected entity or the external url.
```

Internally the `Link` uses a specific overlay depending on the selected type. E.g. medias use the `MediaLinkTypeOverlay`.  

As soon as a new type is selected, the type specific overlay is opened and the user can select the entity or insert 
the external url. Furthermore, it is configurable if setting an `anchor` or a `target` for the resulting url, 

> is allowed in the overlay.

Reviewer: ```suggestion
the external url. Furthermore, it is configurable if setting an `anchor` or a `target` should be possible in the overlay.
```
